quipucords/api/compat/db.py

```diff
@@ -0,0 +1,65 @@
+"""db compatibility module."""
+
+from django.contrib.postgres.aggregates import StringAgg as PostgresStringAgg
+from django.db import connection
+from django.db.models import Aggregate, CharField
+
+
+class SQLiteStringAgg(Aggregate):  # pylint: disable=abstract-method # false positive
+    """StringAgg for sqlite based on GROUP_CONCAT function.
+
+    Aims to be compatible with StringAgg for postgres.
+
+    https://www.sqlite.org/lang_aggfunc.html
+    """
+
+    function = "GROUP_CONCAT"
+    template = "%(function)s(%(expressions)s,%(delimiter)s)"
+
+    def __init__(self, expressions, delimiter=",", **extra):
+        """Initialize aggregator."""
+        self.default_value = self._get_default_value(extra)
+        super().__init__(
+            expressions,
+            delimiter=f'"{delimiter}"',
+            output_field=CharField(),
+            **extra,
+        )
+
+    def convert_value(
+        self, value, expression, connection
+    ):  # pylint: disable=arguments-differ,unused-argument,redefined-outer-name
+        """Convert NULL value to default_value."""
+        # arguments-differ is a false positive here - see how django sql compiler
+        # handles this https://github.com/django/django/blob/f6f0699d01f5840437bfd236c76c797943ef8edc/django/db/models/sql/compiler.py#L1105-L1124 # noqa: 501
+        # other checks were disable to respect the expected signature of this function
+        if value is None:
+            return self.default_value
+        return value
+
+    def _get_default_value(self, extra):
+        default = extra.pop("default")
+        if default is not None:
+            try:
+                # assuming this is an instance of django.db.models.Value
+                return default.value
+            except AttributeError:
+                # ok then... it's not ¯\_(ツ)_/¯
+                return default
+        return None
+
+
+class StringAgg:
+    """Proxy for the appropriate StringAgg based on current db connection."""
+
+    def __new__(cls, expressions, delimiter=",", **extra):
+        """Initialize the proper StringAgg."""
```

Comment:
I didn't quite understand where the connection.vendor information comes from. I understand that __new__ will control the creation of a new instance that doesn't exist up to this point, unlike  the __init__ method. But where exactly do you tell the class where exactly to look for the connection info ?

Reply:
`connection.vendor` info comes from current db connection. 
see line 4 and lines 57 to 60 in this module.

Reply:
Thanks for taking the time to explain. I get it now.